skeleton/cli-bundler/aurelia_project/tasks__if_babel/transpile.js

```diff
@@ -49,13 +62,30 @@ export function buildPluginJavaScript(dest, format) {
     if (format === 'commonjs') {
       opts = {
         plugins: [
-          ['@babel/plugin-transform-modules-commonjs', {loose: true}]
+          ['@babel/plugin-transform-modules-commonjs', {
+            loose: true
+          }]
         ]
       };
     }
     const transpile = babel(opts);
+
+    let defaultBuildOptions = {
+      minify: 'stage & prod',
+      sourcemaps: 'dev & stage',
+      rev: false
+    };
+
+    let buildOptions = new Configuration(project.build.options, defaultBuildOptions);
+    let sourcemaps = buildOptions.isApplicable('sourcemaps');
+
     return gulp.src(project.plugin.source.js)
+      .pipe(gulpIf(sourcemaps, gulpSourcemaps.init()))
       .pipe(transpile)
+      .pipe(gulpIf(sourcemaps, gulpSourcemaps.write('.', {
+        includeContent: false,
+        sourceRoot: '../../src'
```

Comment:
Is `'../../src'` only for the use case of your linked npm package debug?

That sourceRoot seems irrelevant and also harmful to general use case.

Reply:
As I said before, `.pipe(gulp.dest('dist', {sourcemaps: '.'})` seems working fine with default included source contents. Some trouble with your debug session on that simpler setup?

Reply:
if you inline the sources, you are no more able to modify the original source files while debugging

Reply:
Oh I see.

Reply:
Also forgot.

Please add a tailing `/` to `sourceRoot: '../../src/'`. Based on sourcemap spec, "This value is prepended to the individual entries in the “source” field.".

I had encountered some libs before that strictly follow the spec, it leads to `../../srcsource/file/path` without probably inserting `/`.

The tailing `/` can avoid those subtle bugs.

Reply:
There is another small problem.

The package.json has
```
"files": [
    "dist"
  ],
```

It means by default, it only publish dist folder to npm package, not the src folder. We need to update package.json if we don't have embedded source in sourcemaps.

```
"files": [
    "dist",
    "src"
  ],
```